server/src/main/java/com/linecorp/centraldogma/server/internal/storage/repository/git/rocksdb/EncryptionGitStorage.java

```diff
@@ -94,18 +97,27 @@ ObjectId insertObject(ObjectId objectId, int type, byte[] data, int off, int len
         }
 
         final byte[] nonce = AesGcmSivCipher.generateNonce();
-        final byte[] nonceAndType = new byte[NONCE_SIZE_BYTES + 4];
+        // Generate a new DEK for the data so that we don't decrypt and encrypt the data again when the
+        // repository key is rotated.
+        final byte[] objectDek = AesGcmSivCipher.generateAes256Key();
+        final byte[] objectWdek = encrypt(nonce, objectDek, 0, KEY_SIZE_BYTES);
 
-        System.arraycopy(nonce, 0, nonceAndType, 0, NONCE_SIZE_BYTES);
+        assert objectWdek.length == KEY_SIZE_BYTES + 16; // 16 bytes for the tag
 
-        nonceAndType[NONCE_SIZE_BYTES] = (byte) (type >> 24);
-        nonceAndType[NONCE_SIZE_BYTES + 1] = (byte) (type >> 16);
-        nonceAndType[NONCE_SIZE_BYTES + 2] = (byte) (type >> 8);
-        nonceAndType[NONCE_SIZE_BYTES + 3] = (byte) type;
+        final byte[] nonceAndObjectWdek = new byte[NONCE_SIZE_BYTES + 4 + objectWdek.length];
+
+        System.arraycopy(nonce, 0, nonceAndObjectWdek, 0, NONCE_SIZE_BYTES);
+        nonceAndObjectWdek[NONCE_SIZE_BYTES] = (byte) (type >> 24);
+        nonceAndObjectWdek[NONCE_SIZE_BYTES + 1] = (byte) (type >> 16);
+        nonceAndObjectWdek[NONCE_SIZE_BYTES + 2] = (byte) (type >> 8);
+        nonceAndObjectWdek[NONCE_SIZE_BYTES + 3] = (byte) type;
+        System.arraycopy(objectWdek, 0, nonceAndObjectWdek, NONCE_SIZE_BYTES + 4, objectWdek.length);
+
+        final SecretKeySpec keySpec = aesSecretKey(objectDek);
 
         final byte[] encryptedId = encryptObjectId(nonce, objectId);
```

Comment:
Question) If the objective is to ensure easier rotation, does it make sense that both the id and value is encrypted with the `objectWdek`?

i.e. If the `EncryptionGitStorage#dek` is modified, will all objects need to be rewritten since `encryptObjectId` is encrypted directly with `EncryptionGitStorage#dek`?